Adds variant accepting any ack message (#29586)

nvollmar opened the pull request:
References #29586

nvollmar committed:
Adds variant accepting any ack message (#29586)

johanandren commented on akka-stream/src/main/scala/akka/stream/impl/ActorRefBackpressureSinkStage.scala:
```diff
@@ -40,25 +41,42 @@ import akka.stream.stage._
       var completionSignalled = false
 
-      private def receive(evt: (ActorRef, Any)): Unit = {
+      private def receive(ackMsg: Any)(evt: (ActorRef, Any)): Unit = {
```
Givent that all the other logic is identical I think it would be nice to not duplicate and instead go with a guard covering both, something like `case ackMsg if ackMessage.isEmpty || ackMessage.contains(ackMsg) =>  ....`

nvollmar commented on akka-stream/src/main/scala/akka/stream/impl/ActorRefBackpressureSinkStage.scala:
```diff
@@ -5,22 +5,23 @@
  * INTERNAL API
  */
-@InternalApi private[akka] class ActorRefBackpressureSinkStage[In](
+@InternalApi private[akka] class ActorRefBackpressureSinkStage[In, A: ClassTag](
```
Right, need to clean that up, was playing around with an idea...

johanandren reviewed (commented):
Same in the other scaladocs with the `ackMessage` parameter. Looks good to me except for that.

nvollmar committed:
Improves scaladocs and adds typed java api (#29586)

nvollmar commented:
> Same in the other scaladocs with the `ackMessage` parameter. Looks good to me except for that.
I've updated the scaladocs and added typed java variant.

johanandren reviewed (approved):
LGTM, thanks!

nvollmar commented:
@patriknw may I ping you for a review?

raboof reviewed (approved):
Looks really good! Perhaps add a note to /akka-docs/src/main/paradox/stream/operators/Sink/actorRefWithBackpressure.md and /akka-docs/src/main/paradox/stream/operators/ActorSink/actorRefWithBackpressure.md ?

(this PR implements the sink, we should probably add the same API to `(Actor)Source.actorRefWithBackpressure` as well, but that can be a separate PR)

nvollmar committed:
Extends API documentation (#29586)

johanandren commented:
Thanks @nvollmar !
